src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -2934,6 +2935,18 @@ object FunctionRegistry {
     (v: Code[Any]) => CM.invokePrimitive1(t.str)(v) },
     """
     Convert the argument to a human-readable string representation
-    """, "x" -> "the value to be stringified")(TTHr, stringHr)
 
+    .. code-block:: text
+          :emphasize-lines: 2
+
+          let v = Variant("1", 278653, "A", "T") in str(v)
+          result: "1:278653:A:T"
+    """, "x" -> "Value to be stringified.")(TTHr, stringHr)
+
+  registerDependentCode("json", { () =>
+    val t = TT.t
+    (v: Code[Any]) => CM.invokePrimitive1({(x: Any) => JsonMethods.compact(t.toJSON(x))})(v) },
```

Comment:
let's remove the curly braces as they're unnecessary.

Reply:
I think we can actually write this as just:

```scala
CM.invokePrimitive1((x: Any) => JsonMethods.compact(t.toJSON(x)))
```

does that compile?

Reply:
Not without the `(v)` at the end.

Reply:
I think I was unclear, can we replace
```
(v: Code[Any]) => CM.invokePrimitive1({(x: Any) => JsonMethods.compact(t.toJSON(x))})(v)
```

```
CM.invokePrimitive1(JsonMethods.compact(t.toJSON(_)))
```

Reply:
It won't compile in that form.